Support for GeNN 4 timing

neworderofjamie opened the pull request:
GeNN 4 made the timing system slightly simpler to interact with but also renamed things. If kernel timing is enabled, the time file will contain 'neuronUpdateTime', 'presynapticUpdateTime', 'postsynapticUpdateTime', 'synapseDynamicsTime', 'initTime', 'initSparseTime'

neworderofjamie commented:
@tnowotny - could you check this one over too - once we get this and #105 merged into #101 we should be good to iron out the final weird mac issues and release!

tnowotny commented:
So, let me get this right, even if there is no synapse dynamics code, the kernel timing is available anyhow? Is that why no checks are required any longer?

neworderofjamie commented:
Yeah, the variables are always there just zeroed so you don't need to #define out your code to display timings as you don't get compile errors if you query one that isn't used (a user isn't necessarily going to know WHEN the ``initializeSparse`` kernel is going to be launched for instance). It also means you don't need the kind of logic I removed for dealing with them and simplified processing the resultant data - first row will **always** be neuron update time for example.
